Added Learning model, learnings migration and set associations

pavlochernovyy opened the pull request.

pavlochernovyy committed:
Added Learning model, learnings migration and set associations

arsenslyusarchuk commented on db/migrate/20160526093705_create_learnings.rb:
```diff
@@ -0,0 +1,10 @@
+  def change
+    create_table :learnings do |t|
+    	t.belongs_to :user, index: true, foreign_key: true, unique: true, null: false
+	   	t.belongs_to :course, index: true, foreign_key: true, unique: true, null: false
```
bad lines indentation(alignment)

pavlochernovyy committed:
Change lines indentation(alignment)

I thing that happened because of github embedded UI.
In my local machine alignment displays correctly...

arsenslyusarchuk commented on db/migrate/20160526093705_create_learnings.rb:
```diff
@@ -0,0 +1,10 @@
+class CreateLearnings < ActiveRecord::Migration
+  def change
+    create_table :learnings do |t|
+    	t.belongs_to :user, index: true, foreign_key: true, unique: true, null: false
```
unique constraint should be on combination of `user_id` and `course_id` columns to not let the user subscribe to course twice. 
Otherwise if any record appear with same `user_id` or `course_id` it will raise an exception, which doesn't make sense.
Unique constraint on 1 field is useful for case like `email` where u don't want to have 2 users with same email
